source/03-components/table/table--sortable.es6.ts

```diff
@@ -0,0 +1,295 @@
+    const value1 = getCellValue(isAscending ? thisRow : nextRow, index);
+    const value2 = getCellValue(isAscending ? nextRow : thisRow, index);
+
+    if (!value1 || !value2) {
+      return 0;
+    }
+
+    // Compare numerically if they’re numbers.
+    const number1 = Number(value1);
+    const number2 = Number(value2);
+    if (!Number.isNaN(number1) && !Number.isNaN(number2)) {
+      return number1 - number2;
+    }
+
+    // Otherwise, compare alphabetically based on current user locale.
+    return value1
+      .toString()
+      .localeCompare(value2.toString(), navigator.language, {
+        numeric: true,
+        ignorePunctuation: true,
+      });
+  };
+
+/**
+ * Get an Array of column headers elements belonging directly to the given
+ * table element.
+ * @param {HTMLTableElement} table
+ * @return {HTMLTableCellElement[]}
+ */
+const getColumnHeaders = (table: HTMLTableElement): HTMLTableCellElement[] => {
+  const headers = Array.from(
+    table.querySelectorAll<HTMLTableCellElement>(SORTABLE_HEADER)
+  );
+  return headers.filter(header => header.closest(TABLE) === table);
+};
+
+/**
+ * Update the button label within the given header element, resetting it
+ * to the default state (ready to sort ascending) if it’s no longer sorted.
+ * @param {HTMLTableCellElement} header
+ */
+const updateSortLabel = (header: HTMLTableCellElement): void => {
+  const headerName = header.innerText;
+  const sortedAscending = header.getAttribute(SORTED) === ASCENDING;
+  const isSorted =
+    header.getAttribute(SORTED) === ASCENDING ||
+    header.getAttribute(SORTED) === DESCENDING ||
+    false;
+  const headerLabel = `${headerName}, sortable column, currently ${
+    isSorted
+      ? `${sortedAscending ? `sorted ${ASCENDING}` : `sorted ${DESCENDING}`}`
+      : 'unsorted'
+  }`;
+  const headerButtonLabel = `Click to sort by ${headerName} in ${
+    sortedAscending ? DESCENDING : ASCENDING
+  } order.`;
+  header.setAttribute('aria-label', headerLabel);
+  header.querySelector(SORT_BUTTON)?.setAttribute('title', headerButtonLabel);
+};
+
+/**
+ * Remove the aria-sort attribute on the given header element, and reset the
+ * label and button icon.
+ * @param {HTMLTableCellElement} header
+ */
+const unsetSort = (header: HTMLTableCellElement): void => {
+  header.removeAttribute(SORTED);
+  updateSortLabel(header);
+};
+
+/**
+ * Sort rows either ascending or descending, based on a given header’s
+ * aria-sort attribute.
+ * @param {HTMLTableCellElement} header
+ * @param {boolean} isAscending
+ * @return {boolean}
+ */
+const sortRows = (
+  header: HTMLTableCellElement,
+  isAscending: boolean
+): boolean => {
+  header.setAttribute(SORTED, isAscending === true ? DESCENDING : ASCENDING);
+  updateSortLabel(header);
+
+  const table = header.closest(TABLE);
+  const headerRow = header.parentNode;
+
+  if (!table || !headerRow) {
+    return false;
+  }
+
+  const tbody = table.querySelector('tbody') as HTMLTableSectionElement;
```

Comment:
I think this is unnecessary. Instead, we can simply do:
```ts
const tbody = table.querySelector('tbody');
```
Note that this makes the type of `tbody` possibly `null` (`HTMLTableSectionElement | null`), so we should be doing a check that it isn't `null` somewhere before using it:
```ts
if (!tbody) {
  return false;
}
```